File: plugins/set_value/tests/Python/set_value_test_02_share.py
#!/usr/bin/env python3

# SPDX-FileCopyrightText: 2020 Institute of Bioorganic Chemistry Polish Academy of Science (PSNC)
#
# SPDX-License-Identifier: BSD-3-Clause

from sys import argv, exit
import pdi, yaml
import numpy as np

config_path = argv[1]

with open(config_path, 'r') as config_file:
	try:
		config = yaml.safe_load(config_file)
	except yaml.YAMLError as exc:
		exit(exc)

pdi.init(yaml.dump(config))

#scalars
value_char = pdi.access("value_char", pdi.IN)
print("value_char = " + str(value_char))
assert value_char == 42
pdi.release("value_char")

value_short = pdi.access("value_short", pdi.IN)
print("value_short = " + str(value_short))
assert value_short == 4242
pdi.release("value_short")

value_int = pdi.access("value_int", pdi.IN)
print("value_int = " + str(value_int))
assert value_int == 424242
pdi.release("value_int")

value_long = pdi.access("value_long", pdi.IN)
print("value_long = " + str(value_long))
assert value_long == 424242424242
pdi.release("value_long")

value_float = pdi.access("value_float", pdi.IN)
print("value_float = " + str(value_float))
assert abs(value_float - 3.141592) < 0.00001
pdi.release("value_float")

value_double = pdi.access("value_double", pdi.IN)
print("value_double = " + str(value_double))
assert abs(value_double - 3.14159265) < 0.00000001
pdi.release("value_double")

# arrays
array_char = pdi.access("char_array", pdi.IN)
print("char_array = " + str(array_char))
assert array_char[0] == 0
assert array_char[1] == 1
assert array_char[2] == 2
pdi.release("char_array")

array_short = pdi.access("short_array", pdi.IN)
print("short_array = " + str(array_short))
assert array_short[0] == 3
assert array_short[1] == 4
assert array_short[2] == 5
pdi.release("short_array")

array_int = pdi.access("int_array", pdi.IN)
print("int_array = " + str(array_int))
assert array_int[0] == 6
assert array_int[1] == 7
assert array_int[2] == 8
pdi.release("int_array")

array_long = pdi.access("long_array", pdi.IN)
print("long_array = " + str(array_long))
assert array_long[0] == 9
assert array_long[1] == 10
assert array_long[2] == 11
pdi.release("long_array")

array_float = pdi.access("float_array", pdi.IN)
print("float_array = " + str(array_float))
assert abs(array_float[0] - 1.234567) < 0.00001
assert abs(array_float[1] - 12.34567) < 0.00001
assert abs(array_float[2] - 123.4567) < 0.00001
pdi.release("float_array")

array_double = pdi.access("double_array", pdi.IN)
print("double_array = " + str(array_double))
assert abs(array_double[0] - 1.23456789) < 0.00000001
assert abs(array_double[1] - 12.3456789) < 0.00000001
assert abs(array_double[2] - 123.456789) < 0.00000001
pdi.release("double_array")

File: plugins/set_value/tests/set_value_test_02_share.c
/*
 * SPDX-FileCopyrightText: 2020 Institute of Bioorganic Chemistry Polish Academy of Science (PSNC)
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

#include <assert.h>

#include <paraconf.h>
#include <pdi.h>

int main(int argc, char* argv[])
{
	PDI_init(PC_parse_path(argv[1]));

	// char
	char* value_char;
	PDI_access("value_char", (void**)&value_char, PDI_IN);
	assert(*value_char == 42);
	PDI_release("value_char");

	char* char_array;
	PDI_access("char_array", (void**)&char_array, PDI_IN);
	for (int i = 0; i < 3; i++) {
		assert(char_array[i] == i);
	}
	PDI_release("char_array");

	// short
	short* value_short;
	PDI_access("value_short", (void**)&value_short, PDI_IN);
	assert(*value_short == 4242);
	PDI_release("value_short");

	short* short_array;
	PDI_access("short_array", (void**)&short_array, PDI_IN);
	for (int i = 3; i < 6; i++) {
		assert(short_array[i - 3] == i);
	}
	PDI_release("short_array");

	// int
	int* value_int;
	PDI_access("value_int", (void**)&value_int, PDI_IN);
	assert(*value_int == 424242);
	PDI_release("value_int");

	int* int_array;
	PDI_access("int_array", (void**)&int_array, PDI_IN);
	for (int i = 6; i < 9; i++) {
		assert(int_array[i - 6] == i);
	}
	PDI_release("int_array");

	//long
	long* value_long;
	PDI_access("value_long", (void**)&value_long, PDI_IN);
	assert(*value_long == 424242424242);
	PDI_release("value_long");

	long* long_array;
	PDI_access("long_array", (void**)&long_array, PDI_IN);
	for (int i = 9; i < 12; i++) {
		assert(long_array[i - 9] == i);
	}
	PDI_release("long_array");

	//float
	float* value_float;
	PDI_access("value_float", (void**)&value_float, PDI_IN);
	assert(*value_float == 3.141592f);
	PDI_release("value_float");

	float* float_array;
	PDI_access("float_array", (void**)&float_array, PDI_IN);
	assert(float_array[0] == 1.234567f);
	assert(float_array[1] == 12.34567f);
	assert(float_array[2] == 123.4567f);
	PDI_release("float_array");

	//double
	double* value_double;
	PDI_access("value_double", (void**)&value_double, PDI_IN);
	assert(*value_double == 3.14159265);
	PDI_release("value_double");

	double* double_array;
	PDI_access("double_array", (void**)&double_array, PDI_IN);
	assert(double_array[0] == 1.23456789);
	assert(double_array[1] == 12.3456789);
	assert(double_array[2] == 123.456789);
	PDI_release("double_array");

	PDI_finalize();
}
